linera-views/src/batch.rs

```diff
@@ -108,6 +109,48 @@ impl UnorderedBatch {
             insertions,
         })
     }
+
+    /// From an `UnorderedBatch`, creates a [`UnorderedBatch`] that does contain
+    /// key_prefix_deletions that collide with a `key_prefix_deletions`.
+    /// This requires accessing the database to read them and translate them
+    /// as collection of deletes.
+    pub async fn expand_colliding_delete_prefixes<DB: DeletePrefixExpander>(
+        mut self,
+        db: &DB,
+    ) -> Result<UnorderedBatch, DB::Error> {
+        let mut insert_key = BTreeSet::new();
+        for (key, _) in &self.simple_unordered_batch.insertions {
+            insert_key.insert(key.clone());
+        }
```

Comment:
That does not work because the correct solution is
`let inserted_keys = self.simple_unordered_batch.insertions.iter().map(|x| x.0.clone()).collect::<BTreeSet<_>>();`.